src/Sulu/Component/Rest/ListBuilder/Doctrine/FieldDescriptor/DoctrineJoinDescriptor.php

```diff
@@ -86,15 +86,15 @@ public function getEntityName()
      */
     public function getJoin()
     {
-        return $this->join;
+        return str_replace(['\\', ':'], '_', $this->join);
```

Comment:
as we discussed the issue with doctrine 2.6 we said we should not do this magic thing with a str_replace and oriented more by the normal query builder where the `alias` is a required field for 2.0. add a TODO if this will not be fixed in this PR.

Reply:
The problems we had also with the str_replace was in conditions of e.g. joins where the developer writes: `SuluContactBundle.entityClass = "test:Entity"` it would also replace the `:` in its condition string.

Reply:
Is that a real problem here? Will this really be replaced? From what I see `$this->join` only contains the name of the field, not a value.

Reply:
the `join` contains `SuluContactBundle.id` so yes that should be done also here